src/demo/state.js

```diff
@@ -8,7 +8,12 @@ const initialState = {
   trainer: null,
   trainingIndex: 0,
   uiContainer: null,
-  uiElements: []
+  uiElements: [],
+  headerContainer: null,
+  headerElements: [],
+  footerContainer: null,
+  footerElements: [],
+  iterationCount: 0
```

Comment:
Should we unify this default state with the initial state set in `index.jsx`'s `$(document).ready`?

Reply:
i don't think so -- so far, i've been adding all default state in `state.js` and just having `index.jsx` define state related to the DOM. then, `index.jsx` can hook up any state that depends on the current document, and nothing else needs to care about the DOM (e.g., i'm trying to avoid any `document.getElementById()`-esque logic everywhere except `index.jsx`)

if that makes sense 😂